cli/commands/rewards/deploydistributor.ts

```diff
@@ -9,35 +10,37 @@ export default class DeployMerkleDistributor extends BaseCommand {
 
   static flags = {
     merkleTree: flags.string({ required: true, description: 'JSON file with rewards' }),
-    env: flags.string({ required: false, description: 'blockchain environment with which to interact' }),
+    env: flags.string({ required: false, description: '[default: local] Blockchain environment with which to interact' }),
     from: flags.string({ required: true, description: 'Deployer address' }),  
     privateKey: flags.string({ required: false, description: 'Use a private key to sign local transactions with' }), 
   }
 
   async run() {
     const res = this.parse(DeployMerkleDistributor)
-    const merkleTree = JSON.parse(fs.readFileSync(res.flags.merkleTree, { encoding: 'utf8' }))
+    let merkleTree = JSON.parse(fs.readFileSync(res.flags.merkleTree, { encoding: 'utf8' }))
     const from: string = res.flags.from.toLowerCase()
-    const kit = newKit(this.nodeByEnv(res.flags.env))
-    const stableToken = await kit.contracts.getStableToken()
+    const celoToken = await this.kit.contracts.getGoldToken()
     const abi = MerkleDistributor.abi 
-    if (res.flags.privateKey) {
-      kit.addAccount(res.flags.privateKey)
-    }
 
     // @ts-ignore - unhappy with the abi format, but it is valid
-    let merkleDistributor = new kit.web3.eth.Contract(abi)
-    let contract = await merkleDistributor.deploy({
-        data: MerkleDistributor.bytecode,
-        arguments: [stableToken.address, merkleTree.merkleRoot]
-    }).send({ 
-        from,
-        gas: 1500000,
-        gasPrice: '30000000000000'
-    })
+    let merkleDistributor = new this.kit.web3.eth.Contract(abi)
+    let txResult = await toTransactionObject(
+      this.kit.connection,
+      // @ts-ignore
+      merkleDistributor.deploy({
```

Comment:
@nambrot is there a good way to convert this to a `CeloTxObject`?

Reply:
I think you would need to use typechain to create typings for merkleDistributor in this case but @yorhodes and @m-chrzan would know more